src/components/Modal.svelte

```diff
@@ -10,6 +11,7 @@
     role="button"
     tabindex="0"
     class="fixed right-0 top-0 z-[100] flex h-[100vh] w-[calc(100vw-126px)] cursor-default items-center justify-center bg-overlay"
+    {id}
```

Comment:
Do we know that there only ever one modal at a time? Otherwise we might not want to set an id tag and get collisions.

Reply:
That is a prop that is set outside of the reusable component.  In this PR, I am setting it to a unique value:

`<Modal id="stake-to-provider" {isOpen} {close}>`

However, in ConnectProvider it is not set.  We need to either always set it to a unique value or not set it at all.  Furthermore, the Modal reusable component design is not enforcing modal behavior.  It's strictly visual.  @claireolmstead  can elaborate on it.

Reply:
@rlaferla that makes sense. I'd remove the default value in that case and make setting the Id as a required prop.

Reply:
@wilwade @claireolmstead should make this change in her PR for `ConnectProvider`.

In `Modal.svelte`:

```
<script lang="ts">
  export let close = () => {};
  export let isOpen: boolean = false;
  export let id: string;
</script>
```
In `ConnectProvider.svelte`:

`<Modal id="connect-provider" {close} {isOpen}>
`

Reply:
I'm confused. The id tag is being added here. Why would it be changed in a different PR?

Reply:
@wilwade I was referring to the change to `ConnectProvider`.  I didn't write the `Modal` component but I'll make both changes in this PR.

Reply:
fixed